### PR TITLE
Price of food depends on its nutritional capacity

### LINKED ISSUE #11464: bread slices sell for more than the bread they come from.
## Description

chopping up bread into slices allows you to multiply the value of the bread, since the loaf and each slice sell for the same amount.
**Reproduction**

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by corentt**
## About the PR
This allow to calculate a price based on the reagents in a SolutionContainerManager.

The reagent prototype now have a PricePerUnit datafield that is used is the calculation.
Right now only Nutriment, Vitamin and Protein got a price.

This can easily be extended to drinks and chemicals.

fixes #11464

**Changelog**

:cl:
- add: the price of food now depends on its nutritional capacity

### CODE REVIEW

**Comment by metalgearsloth** [comment]
Kinda weird having SolutionPrice considering the same reagent in one container or another changes its price.

**Comment by corentt** [comment]
> Kinda weird having SolutionPrice considering the same reagent in one container or another changes its price. 

You mean I should remove that component and calculate the price on every SolutionContainer ?

**Comment by metalgearsloth** [comment]
> > Kinda weird having SolutionPrice considering the same reagent in one container or another changes its price.
> 
> You mean I should remove that component and calculate the price on every SolutionContainer ?

yes, it seems like it's too prone to data entry bugs where contributors forget about it.

File: Content.Server/Cargo/Systems/PricingSystem.cs [hunk 1]
```diff
@@ -108,6 +112,25 @@ private void CalculateStackPrice(EntityUid uid, StackPriceComponent component, r
         args.Price += stack.Count * component.Price;
     }
 
+    private void CalculateSolutionPrice(EntityUid uid, SolutionContainerManagerComponent component, ref PriceCalculationEvent args)
+    {
+        var price = 0f;
+
+        foreach( ( _ ,var solution) in component.Solutions)
+        {
```
Comment (metalgearsloth) [inline Content.Server/Cargo/Systems/PricingSystem.cs:120]:
```suggestion
        foreach (var solution in component.Solutions.Values)
        {
```

File: Content.Server/Cargo/Systems/PricingSystem.cs [hunk 2]
```diff
@@ -108,6 +112,25 @@ private void CalculateStackPrice(EntityUid uid, StackPriceComponent component, r
...
+            if (solution.Contents.Count() == 0)
+                continue;
+
+            foreach(var reagent in solution.Contents)
```
Comment (metalgearsloth) [inline Content.Server/Cargo/Systems/PricingSystem.cs:124]:
```suggestion
            foreach (var reagent in solution.Contents)
```

Unnecessary when you're just iterating over it.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**